server/middleware/registerUser.js

```diff
@@ -1,10 +1,24 @@
 const { Users } = require("../database/databaseModel");
+const bcrypt = require("bcrypt");
 
 const registerUser = (request, response, next) => {
   const { username, password, email } = request.body;
 
   Users.create({ username, password, email })
     .then(() => {
+      const saltRounds = 10;
+
+      bcrypt.hash(password, saltRounds, (error, hash) => {
+        Users.update(
+          { password: hash },
+          {
+            where: {
+              username: username,
+            },
+          }
+        );
+      });
+
       return next();
```

Comment:
I don't think there's ever a need to return next()

And I think next is being called in the "main threat" (ish) of this bit of code, while the Users.update runs asynchronously, which means next will be called even if Users.update fails - probably not wanted!